docs/source/tutorials/datasets.rst

```diff
@@ -295,6 +295,21 @@ and create your own class.
     dataset.template=import.path.to.CustomTemplate
 
 
+Under the hood, torchtune is using :code:`importlib.import_module`. 
+You can place your custom template class in any Python module as long as the module is 
+accessible by Python's import mechanism. This means the module should be in a directory 
+that is included in Python's search paths (:code:`sys.path`). 
+This often includes:
+
+- The current directory from which your Python interpreter or script is run.
+- Directories where Python packages are installed (like :code:`site-packages`).
+- Any directories added to :code:`sys.path` at runtime or through the :code:`PYTHONPATH` environment variable.
+
+If you are working within a cloned repository of torchtune you can just define your custom template 
```

Comment:
I know this is a viable option, but I would rather encourage users to create custom components in their own project directories instead of having to modify torchtune internals just to run fine-tuning jobs. From a library maintainer's perspective, this doesn't seem like a good user flow - they should be able to import our library and make their custom scripts and building blocks using our components - just like with core PyTorch. And if there's anything that can be improved to make that workflow easier then we need to do that. What do you think?

Reply:
I agree, modifying library internals should be only for maintainers and contributors. 
Should we phrase it like : create your own custom dir that contains your configs and custom components you want to link to torchtune ? 
But this would deserve I whole walkthrough because there are different level of complexity : from the user that just as a custom dataset from an user that has a whole new model or loss function

Reply:
You could say "In your own repository, define your custom template in its own file and you can then access directly from your config by providing the import path"